pkg/event-consumers/kafka/kafka-consumer.go

```diff
@@ -214,20 +216,29 @@ func (c *Consumer) Cleanup(sarama.ConsumerGroupSession) error {
 
 // ConsumeClaim must start a consumer loop of ConsumerGroupClaim's Messages().
 func (c *Consumer) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
+	b := backoff.NewExponentialBackOff()
+	b.MaxElapsedTime = 0 // ... so that b.NextBackOff() never returns backoff.Stop.
+	b.MaxInterval = 60 * time.Second
+
 	for msg := range claim.Messages() {
 		req, err := utils.GetHTTPReq(c.funcName, c.funcPort, msg.Topic, c.ns, kafkatriggersNamespace, "POST", string(msg.Value))
 		if err != nil {
 			logrus.Errorf("Unable to elaborate request (namespace = %v function = %v topic = %v partition = %v offset = %v): %v", c.ns, c.funcName, msg.Topic, msg.Partition, msg.Offset, err)
 			continue
 		}
 
-		if err = utils.SendMessage(req); err != nil {
-			logrus.Errorf("Failed to send message (namespace = %v function = %v topic = %v partition = %v offset = %v): %v", c.ns, c.funcName, msg.Topic, msg.Partition, msg.Offset, err)
-		} else {
-			logrus.Infof("Message sent successfully (namespace = %v function = %v topic = %v partition = %v offset = %v)", c.ns, c.funcName, msg.Topic, msg.Partition, msg.Offset)
+		err = utils.SendMessage(req)
+		session.MarkMessage(msg, "")
+
+		if err != nil {
+			d := b.NextBackOff()
```

Comment:
I would make this feature optional, note that people may still prefer the old behavior. For example if one message is sent every 10 minutes, they may want to to execute the function at that moment, regardless of its result.

Reply:
Good point 👍 
I've added environment variable `BACKOFF_INTERVAL`, which enables the feature.